### PR TITLE
Use respect_dns_ttl in bootstrap

### PR DESCRIPTION
**Pull request by howardjohn**
For https://github.com/istio/istio/issues/21222

This cleans up our config a bit. Rather than resolving DNS every 5 min for pilot, just use the TTL return in the dns response. This is what we are doing for ServiceEntries already, so its proven to work fine.

There is 1000s of lines of autogenerated garbage, see the first commit for easier review, the second one just adds the gen

### CODE REVIEW

File: pilot/cmd/pilot-agent/main.go [hunk 1]
```diff
@@ -808,8 +806,6 @@ func init() {
 	// See https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-component-log-level
 	proxyCmd.PersistentFlags().StringVar(&proxyComponentLogLevel, "proxyComponentLogLevel", "misc:error",
 		"The component log level used to start the Envoy proxy")
-	proxyCmd.PersistentFlags().StringVar(&dnsRefreshRate, "dnsRefreshRate", "300s",
```
Comment (hzxuzhonghu) [inline pilot/cmd/pilot-agent/main.go]: Would suggest first deprecate it rather than removing directly

**Review by hzxuzhonghu — COMMENTED** [review]
Otherwise, lgtm

File: pilot/cmd/pilot-agent/main.go (hunk 1 above)
Comment (howardjohn) [reply pilot/cmd/pilot-agent/main.go]: why? we support unknown flags now, so I am not sure what the use case is for setting this flag. ServiceEntry has been behaving this way for 2 releases now

Comment (hzxuzhonghu) [reply pilot/cmd/pilot-agent/main.go]: Oh, i did not know that. Then it should work.

**Review by hzxuzhonghu — APPROVED** [review]

File: pilot/cmd/pilot-agent/main.go (hunk 1 above)
Comment (howardjohn) [reply pilot/cmd/pilot-agent/main.go]: Yeah we added it a while back for similar reason

**Review by ramaraochavali — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Review by richardwxn — APPROVED** [review]

**Pull request merged**